bot/cogs/define_cog.py

```diff
@@ -0,0 +1,150 @@
+                pronunc = []
+                try:
+                    prsData = wordData['hwi']['prs']
+                    for prs in range(len(prsData)):
+                        soundData = prsData[i]
+                        pronunc.append(soundData['mw'])
+
+                except:
+                    pass
+                
+                #Type of the given word (Noun, Verb, Adjective, etc.)
+                wordType = ""
+                try:
+                    wordType = wordData['fl']
+                except:
+                    pass
+                
+                #Definitions of the given word
+                definitions = []
+                try:
+                    defData = wordData['shortdef']
+                    for defin in range(len(defData)):
+                        definitions.append(defData[defin])
+                except:
+                    definitions.append("Definition Missing")
+                
+                #Turn data into one long string (represents a page)
+                template = "Tenses: "
+                for s in range(len(wordStems)):
+                    template += wordStems[s]
+                    if (s != len(wordStems) - 1):
+                        template += ", "
+                template += "\n"
+
+                template += "Syllables: " + syllableData + "\n"
+
+                template += "Pronunciation: "
+                for s in range(len(pronunc)):
+                    template += pronunc[s]
+                    if (s != len(pronunc) - 1):
+                        template += ", "
+                template += "\n"
+
+                template += "Word Type: " + wordType + "\n"
+
+                template += "\n"
+                for s in range(len(definitions)):
+                    page = template + "Definition: " + definitions[s]
+                    page = page.replace("*"," | ")
+                    pages.append(page)
+
+        #If the word cannot be found, the JSON returns a list of other possible suggestions. 
+        elif (isinstance(jsonData[0], str)):
+            template = "Word not found, see also: "
+            for s in range(len(jsonData)):
+                template += jsonData[s]
+                if (s != len(jsonData) - 1):
+                    template += ", "
+    
+            pages = [template]
+
+        return pages
+    
+    @commands.command()
+    async def define(self, ctx, word):
+        """
+        Given a word, find its definition and any other relevant information
+        
+        USE: !define <word>
+        EXAMPLE: !define schadenfreude
+
+        For phrases, use underscores
+        EXAMPLE: !define computer_science
+        """
+
+        actualWord = word.replace("_"," ")
+        word = word.replace("_","%20")
+        url = "https://www.dictionaryapi.com/api/v3/references/collegiate/json/" + word.lower() + "?key=" + self.api_key
+        response = requests.get(url)
```

Comment:
we cant use the requests module in Clembot, this is a blocking call, this call completely shuts down the bot for the duration of the request. We have to use aiohttp which is the async version of http requests.  https://docs.aiohttp.org/en/stable/